pkg/destroy/azure/azure.go

```diff
@@ -193,10 +193,22 @@ func (o *ClusterUninstaller) Run() error {
 	return utilerrors.NewAggregate(errs)
 }
 
-func deletePublicRecords(ctx context.Context, dnsClient dns.ZonesClient, recordsClient dns.RecordSetsClient, privateDNSClient privatedns.PrivateZonesClient, privateRecordsClient privatedns.RecordSetsClient, logger logrus.FieldLogger, rgName string) error {
+func deletePublicRecords(ctx context.Context, rgClient resources.GroupsClient, dnsClient dns.ZonesClient,
+	recordsClient dns.RecordSetsClient, privateDNSClient privatedns.PrivateZonesClient,
+	privateRecordsClient privatedns.RecordSetsClient, logger logrus.FieldLogger, rgName string) error {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
 	defer cancel()
 
+	// check if resource group exists
+	resourceGroup, err := rgClient.Get(ctx, rgName)
+	if err != nil {
+		if resourceGroup.Response.StatusCode == http.StatusNotFound {
+			logger.Debug("resource group already deleted, no records to delete")
+			return nil
+		}
+		return err
+	}
```

Comment:
instead of making another call, check for no-work error here https://github.com/openshift/installer/pull/4325/files#diff-a2fdf21d9b56bf657a0251ad81376851d27e5b801510337c8f5c9d2cd6495885R216-R221

Reply:
I think the link points to this commit diff.. Can you tell me where to check?

Reply:
Addressed.
There is a type mismatch to use `wasNotFound` here, but I found an inbuilt method that essentially does the same thing